vendor/github.com/containers/image/docker/docker_client.go

```diff
@@ -146,21 +156,22 @@ func (c *dockerClient) setupRequestAuth(req *http.Request) error {
 			// no need for bearer? wtf?
 			return nil
 		}
-		tokens = strings.Split(hdr, " ")
-		tokens = strings.Split(tokens[1], ",")
-		var realm, service, scope string
-		for _, token := range tokens {
-			if strings.HasPrefix(token, "realm") {
-				realm = strings.Trim(token[len("realm="):], "\"")
-			}
-			if strings.HasPrefix(token, "service") {
-				service = strings.Trim(token[len("service="):], "\"")
-			}
-			if strings.HasPrefix(token, "scope") {
-				scope = strings.Trim(token[len("scope="):], "\"")
-			}
+		re := regexp.MustCompile(`^Bearer realm="(.*)",service="(.*)",scope="(.*)"$`)
```

Comment:
> Actually, we do have integration tests pulling from Docker Hub, which does test this code path.

Turns out we only test pull, i.e. not the part that broke, and pushing to OpenShift, which does not use `scope`s at all, does not exercise this either.

So, it would still be nice to have an integration test which pushes to `docker/distribution`, considering we are building three versions of that in the `Dockerfile`…